### PR TITLE
docs: Enforce Markdownlint for doc uniformity

### PR DESCRIPTION
**Pull request by hcgatewood**
## Summary

Docs are often messy and non-uniform, and reviewing these trivialities is not worth the manual effort. This PR introduces [markdownlint](https://github.com/DavidAnson/markdownlint) as a (soon-to-be required) CI check, calling the accompanying precommit command.

This PR also fixes all 800 existing lint issues for the full set of Docusaurus readmes, and also removes (or promotes) legacy documentation.

## Test Plan

Screen Shot 2021-07-31 at 1 19 00 AM

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by themarwhal — APPROVED** [review]
Awesome!

**Review by rdefosse — APPROVED** [review]

**Review by electronjoe — APPROVED** [review]
I hope the tool was able to correct the 800 lint findings for you =D

File: .github/workflows/docs-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+      - name: Run docs precommit
+        run: |
+            cd ${MAGMA_ROOT}/docs
+            make precommit
+#      - name: Extract commit title
```
Comment (quentinDERORY) [inline .github/workflows/docs-workflow.yml]: Can you get rid of the commented code. We can always add it later.
(And I have a pending PR updating this logic...)

**Comment by hcgatewood** [comment]
> I hope the tool was able to correct the 800 lint findings for you =D

HA. I fixed a fair few 🙂

**Pull request merged**